pkg/services/pluginsintegration/plugininstaller/service.go

```diff
@@ -108,6 +108,10 @@ func (s *Service) shouldUpdate(ctx context.Context, pluginID, currentVersion str
 		return false
 	}
 
+	// Private plugins won't have entry in our catalog
+	if info == nil || info.Version == "" {
```

Comment:
`GetPluginArchiveInfo` -> calls the gcom `/api/plugins/:pluginId/versions` that will throw `ErrResponse4xx` for private plugins since it does not exist.